Update some naming details in documentation (#25735)
DataDog/datadog-agent: docs/public/components/creating-components.md

    package implzstd

    // NewComponent returns a new ZSTD implementation for the compression component
    func NewComponent(reqs Requires) Provides {
        ....
    }
    ```

To require input arguments to the `NewComponent` instantiation function, use a special struct named `Requires`.
The instantiation function returns a special stuct named `Provides`. This internal nomenclature is used
to handle the different component dependencies using Fx groups.

> In this example, the compression component must access the configuration component and the log component. To express this, define a `Requires` struct with two fields. The name of these field is irrelevant, but the type must be the concrete type of the interface you require.

Reviewer: ```suggestion
In this example, the compression component must access the configuration component and the log component. To express this, define a `Requires` struct with two fields. The name of the fields is irrelevant, but the type must be the concrete type of interface that you require.
```
